src/App.jsx

```diff
@@ -1,22 +1,51 @@
-import React, { useEffect } from "react";
+import { useState, useEffect } from "react";
 import TodoList from "./TodoList";
 import TodoForm from "./AddTodoForm";
 
-// Custom hook to manage semi-persistent state using localStorage
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = React.useState(
-    JSON.parse(localStorage.getItem("savedTodoList")) || []
-  );
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  // Fetch data from Airtable
+  async function fetchData() {
```

Comment:
I don't think you need to change this for this assignment, but I thought I'd mention a style/philosophical point on your fetch function.

This function is taking up a lot of space in your `App` component, so it's worth trying to pull it out.  Then you have a decision about whether it should be a function that only gets this type of data, or about whether you want to create an abstraction that can be used to get any kind of data.

You might have a base method `httpRequest`, then a method `fetchData` that calls `httpRequest` with `method: "GET"`, then `fetchDataFromAirtable` that calls `fetchData` but with the url and bearer token set, etc.  This way you're creating reusable functions that will allow you to avoid rewriting code then next time you have to perform a similar task.  Probably they would all live in a `RequestHandler` class or something.

Keep in mind that in order for this request abstraction to live outside of your React component, you would need to pass the state setters `setTodoList`, `setIsLoading` in as arguments.  Let me know if you want to discuss more about this.  It's beyond the scope of this lesson, but thinking in abstractions is an important part of writing professional code.